src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -109,80 +137,116 @@ private ReadFilterLibrary(){ /*no instance*/ }
+    /** Keep only paired reads (0x1) that are second of pair (0x80). */
+    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Keep only paired reads that are second of pair")
     public static class SecondOfPairReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test (final GATKRead read) {
             return read.isSecondOfPair();}}
 
-    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
+    /** Keep reads with sequenced bases. */
+    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Keep reads with sequenced bases")
     public static class SeqIsStoredReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return read.getLength() > 0;}}
 
-    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
+    /**
+     * Keep reads with a valid alignment start (POS larger than 0) or is unmapped.
+     * See MappedReadFilter for criteria defining an unmapped read.
+     */
+    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Keep reads with a valid alignment start", extraDocs = MappedReadFilter.class)
     public static class ValidAlignmentStartReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return read.isUnmapped() || read.getStart() > 0;}}
 
-    // Alignment doesn't align to a negative number of bases in the reference.
-    //Note: to match gatk3 we must keep reads that align to zero bases in the reference.
-    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
+    /**
+     * Keep reads where the read end corresponds to a proper alignment -- that is, the read ends after the start
+     * (non-negative number of bases in the reference), calculated as:
+     *
+     * <code>
+     * end - start + 1 &ge; 0
+     * where
+     *  start = 1-based inclusive leftmost position of the clipped sequence (0 if no position)
+     *  end = 1-based inclusive rightmost position of the clipped sequence (0 if unmapped)
+     * </code>
+     *
+     * <p>Note: keep also unmapped reads (align to zero bases in the reference). See MappedReadFilter for criteria defining an unmapped read.
+     */
+    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Keep reads where the read end correspond to a proper alignment", extraDocs = MappedReadFilter.class)
```

Comment:
"correspond to a proper alignment" -> "is properly aligned" is more straightforward